File: csharp/side exercises/series/Series.cs
using System;
using System.Collections.Generic;

public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength > numbers.Length)
            throw new ArgumentException();

        if (sliceLength <= 0)
            throw new ArgumentException();

        if (numbers.Length == 0)
            throw new ArgumentException();

        List<string> series = new List<string>();

        for (int sliceStart = 0; sliceStart <= numbers.Length - sliceLength; sliceStart++)
            series.Add(numbers.Substring(sliceStart, sliceLength));

        return series.ToArray();
    }
}

File: java/side exercises/series/src/main/java/Series.java
import java.util.List;
import java.util.ArrayList;

class Series {
	
	private String number;
	
	Series (String number){
		this.number = number;
	}
	
	List<String> slices (int slice){
		
		if (slice < 1) {
			throw new IllegalArgumentException ("Slice size is too small.");
		}
		
		if (slice > number.length()) {
			throw new IllegalArgumentException ("Slice size is too big.");
		}
		
		List<String> result = new ArrayList<>();
		
		if (slice == number.length()) {
			result.add(number);
		} else {
			for (int i = 0; i < number.length() - slice + 1; i++){
				result.add(number.substring (i, i + slice));
			}
		}
		
		return result;
	}
}
